### PR TITLE
Kernel/Net: E1000ENetworkAdapter detect EEPROM

### PR DESCRIPTION
**Pull request by TomFinet**
The presence of EEPROM is detected by reading the EE_PRES bit from the EECD register.

### CODE REVIEW

File: Kernel/Net/Intel/E1000ENetworkAdapter.cpp [hunk 1]
```diff
@@ -235,19 +241,18 @@ UNMAP_AFTER_INIT E1000ENetworkAdapter::E1000ENetworkAdapter(StringView interface
     NonnullOwnPtr<Memory::Region> tx_buffer_region, NonnullOwnPtr<Memory::Region> rx_descriptors_region,
     NonnullOwnPtr<Memory::Region> tx_descriptors_region)
     : E1000NetworkAdapter(interface_name, device_identifier, irq, move(registers_io_window),
-        move(rx_buffer_region),
```
Comment (LucasChollet) [inline Kernel/Net/Intel/E1000ENetworkAdapter.cpp:238]: Please avoid changing unrelated code.

File: Kernel/Net/Intel/E1000ENetworkAdapter.cpp [hunk 2]
```diff
@@ -235,19 +241,18 @@ UNMAP_AFTER_INIT E1000ENetworkAdapter::E1000ENetworkAdapter(StringView interface
...
 {
 }
 
 UNMAP_AFTER_INIT E1000ENetworkAdapter::~E1000ENetworkAdapter() = default;
 
 UNMAP_AFTER_INIT void E1000ENetworkAdapter::detect_eeprom()
 {
-    // FIXME: Try to find a way to detect if EEPROM exists instead of assuming it is
-    m_has_eeprom = true;
+    m_has_eeprom = in32(REG_EECD) & EECD_PRES;
```
Comment (LucasChollet) [inline Kernel/Net/Intel/E1000ENetworkAdapter.cpp:255]: It would be nice to have more details on that. Is there a particular section in the specification?

**Review by LucasChollet — COMMENTED** [review]
Small nit on the commit title:
`Kernel/Net` would be preferred.

**Review by LucasChollet — APPROVED** [review]

**Review by ADKaster — APPROVED** [review]
